plugins/collapseGroups.js

```diff
@@ -80,23 +108,17 @@ export const fn = () => {
                 node.attributes.transform == null &&
                 firstChild.attributes.transform == null))
           ) {
+            if (!canMoveAttributesToSingleChild(node, firstChild)) {
+              return;
+            }
             for (const [name, value] of Object.entries(node.attributes)) {
```

Comment:
Instead of potentially iterating twice, I think it'd be better to shallow copy the attributes and assign them if we complete the loop without returning.

This achieves the same thing while being fewer lines of code.

```js
const newChildElemAttrs = { ...firstChild.attributes };

for (const [name, value] of Object.entries(node.attributes)) {
  // avoid copying to not conflict with animated attribute
  if (hasAnimatedAttr(firstChild, name)) {
    return;
  }

  if (newChildElemAttrs[name] == null) {
    newChildElemAttrs[name] = value;
  } else if (name === 'transform') {
    newChildElemAttrs[name] = value + ' ' + newChildElemAttrs[name];
  } else if (newChildElemAttrs[name] === 'inherit') {
    newChildElemAttrs[name] = value;
  } else if (!inheritableAttrs.has(name) && newChildElemAttrs[name] !== value) {
    return;
  }
}

node.attributes = {};
firstChild.attributes = newChildElemAttrs;
```

Reply:
Made the change and re-ran regression.js locally, same results as above.

The reason my regression numbers are different is that I'm running on Windows, and there seem to be a small number of (consistent) false positives. Every one I've looked at so far has been something in a `<text>` element which looks like a minor antialiasing issue. Hopefully I'll figure out a solution to this at some point.

Reply:
Thanks for the explanation. It helps (and sucks) to know different operating systems, or possibly different hardware(?), have an impact on our tests.